code/modules/dwarven/dwarven_items.dm

```diff
@@ -0,0 +1,244 @@
+		if(AM == user || AM.anchored)
+			continue
+
+		if(ismob(AM))
+			var/mob/M = AM
+			if(M.anti_magic_check())
+				continue
+
+		throwtarget = get_edge_target_turf(user, get_dir(user, get_step_away(AM, user)))
+		distfromcaster = get_dist(user, AM)
+		if(distfromcaster == 0)
+			if(isliving(AM))
+				var/mob/living/M = AM
+				M.Paralyze(100)
+				M.adjustBruteLoss(5)
+				to_chat(M, "<span class='userdanger'>You're slammed into the floor by [user]!</span>")
+		else
+			if(isliving(AM))
+				var/mob/living/M = AM
+				M.Paralyze(stun_amt)
+				to_chat(M, "<span class='userdanger'>You're thrown back by [user]!</span>")
+			AM.safe_throw_at(throwtarget, ((clamp((maxthrow - (clamp(distfromcaster - 2, 0, distfromcaster))), 3, maxthrow))), 1,user, force = MOVE_FORCE_EXTREMELY_STRONG)//So stuff gets tossed around at the same time.
+	..()
+
+/obj/item/dwarven/rune_stone/air
+	name = "air runestone"
+	overlay_state = "air_rune"
+
+/obj/item/dwarven/rune_stone/air/apply(atom/target, mob/user)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] fizzles on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		var/teleammount = 0
+		var/teleloc = target
+		if(!isturf(target))
+			teleloc = target.loc
+		for(var/atom/movable/stuff in teleloc)
+			if(!stuff.anchored && stuff.loc && !isobserver(stuff))
+				if(do_teleport(stuff, stuff, 10, channel = TELEPORT_CHANNEL_MAGIC))
+					teleammount++
+					var/datum/effect_system/smoke_spread/smoke = new
+					smoke.set_up(max(round(4 - teleammount),0), stuff.loc) //Smoke drops off if a lot of stuff is moved for the sake of sanity
+					smoke.start()
+	..()
+
+/obj/item/dwarven/mold
+	name = "dwarven mold"
+	desc = "Dwarven mold, one of their great achievments. Allows for casting of very complex tools and armors"
+	icon_state = "mold"
+	w_class = WEIGHT_CLASS_SMALL
+	var/mold_type
+
+/obj/item/dwarven/mallet
+	name = "dwarven mallet"
+	desc = "Dwarven mallet, easy to make , easy to use, other than the fact it is absolutely tiny"
+	icon_state = "mallet"
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/dwarven/blueprint
+	name = "dwarven structure print"
+	desc = "Dwarven instructions on how to build a dwarven structure, includes materials how neat."
+	icon_state = "structure_print"
+	w_class = WEIGHT_CLASS_SMALL
+	var/obj/structure/destructible/dwarven/structure
+
+/obj/item/dwarven/blueprint/New(loc,_structure)
+	structure = _structure
+	. = ..()
+
+/obj/item/dwarven/blueprint/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!proximity_flag)
+		return
+	if(isclosedturf(target) ||  isgroundlessturf(target))
+		return FALSE
+	if(!do_after(user, 60, TRUE, user))
+		return FALSE
+	var/turf/place = isopenturf(target) ? target : get_turf(target)
+	new structure(place)
+	qdel(src)
+	. = ..()
+
+/obj/item/dwarven/upgrade_kit
+	name = "dwarven modification kit"
+	desc = "Dwarven instructions on how to modify a weapon."
```

Comment:
game was broke enough to where i didn't get far enough but same thing with adding a description on how to use